core/src/main/scala/org/apache/spark/deploy/worker/Worker.scala

```diff
@@ -220,6 +225,38 @@ private[deploy] class Worker(
     metricsSystem.getServletHandlers.foreach(webUi.attachHandler)
   }
 
+  // TODO if we're starting up multi workers under the same host, discovery script won't work.
+  private def setupWorkerResources(): Unit = {
+    try {
+      resources = resourceFile.map { rFile =>
+        ResourceDiscoverer.parseAllocatedFromJsonFile(rFile)
+      }.getOrElse {
+        if (resourceDiscoveryScript.isEmpty) {
```

Comment:
We don't want to reuse the discoveryScript config?

Reply:
Not sure what you actually mean here. Just explain my understanding. `resourceDiscoveryScript` is configured from `SPARK_WORKER_RESOURCE_DISCOVERY_SCRIPT` or `--resource-script`, which are specially for Worker. And, application(driver/executor) discoveryScipt config haven't setup while Worker is starting up.